deploy/7-deploy-pendle-oracle.ts

```diff
@@ -4,49 +4,119 @@ import { HardhatRuntimeEnvironment } from "hardhat/types";
 
 import { ADDRESSES } from "../helpers/deploymentConfig";
 
+type OracleConfig = {
+  name: string;
+  market: string;
+  token: string;
+  collateral: string;
+  customOracle: string;
+}[];
+
+const deployPendleOracle = async (
+  deployments: HardhatRuntimeEnvironment["deployments"],
+  deployer: string,
+  name: string,
+  args: any[],
+  proxyOwnerAddress: string,
+  defaultProxyAdmin: any,
+) => {
+  const { deploy } = deployments;
+  return deploy(name, {
+    contract: "PendleOracle",
+    from: deployer,
+    log: true,
+    deterministicDeployment: false,
+    args,
+    proxy: {
+      owner: proxyOwnerAddress,
+      proxyContract: "OptimizedTransparentUpgradeableProxy",
+      viaAdminContract: {
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
+    },
+    skipIfAlreadyDeployed: true,
+  });
+};
+
 const func: DeployFunction = async ({
   getNamedAccounts,
   deployments,
   network,
   artifacts,
 }: HardhatRuntimeEnvironment) => {
-  const { deploy } = deployments;
   const { deployer } = await getNamedAccounts();
 
   const oracle = await ethers.getContract("ResilientOracle");
   const proxyOwnerAddress = network.live ? ADDRESSES[network.name].timelock : deployer;
-
   const defaultProxyAdmin = await artifacts.readArtifact(
     "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
   );
 
-  const { PTweETH_26DEC2024, PTweETH_26DEC2024_Market, PTOracle, WETH } = ADDRESSES[network.name];
+  const addresses = ADDRESSES[network.name];
+  const fallbackAddress = "0x0000000000000000000000000000000000000001";
 
-  const ptOracleAddress = PTOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+  const deployOracles = async (oracleConfig: OracleConfig) => {
+    for (const { name, market, token, collateral, customOracle } of oracleConfig) {
+      const ptOracleAddress = customOracle || (await ethers.getContract("MockPendlePtOracle")).address;
+      await deployPendleOracle(
+        deployments,
+        deployer,
+        name,
+        [market || fallbackAddress, ptOracleAddress, token, collateral, oracle.address, 1800],
```

Comment:
The `twapDuration` can also change, depending on the asset. I would move it to the  `oracleConfig` configuration